src/components/Modal/README.md

```diff
@@ -243,6 +276,13 @@ export default {
 	inject: {
 		modalApi,
 	},
+
+	methods: {
+		beforeCloseHook() {
```

Comment:
Might be good to set the default recommendation to async since the most common use-case would either be to dispatch an async save action or confirmation dialog.

```suggestion
		async beforeCloseHook() {
```